js/algorithms/simple-arbitrage.js

```diff
@@ -1,63 +1,78 @@
-function outputExchangeData(allExchangeData) {
+function outputExchangeData(pair, time) {
 
   // Temporary variables for algorithm
   var highest_bid = 0;
   var highest_bid_amt = 0;
   var lowest_ask = 10000000000;
   var lowest_ask_amt = 0;
   var low_amt;
+  var data = {};
+
+  // Passing in subset constructor
+  var subSet = require("./../classes/sub-set.js");
+
+  // Creating a subset of data and opening feeds to that data
+  var Exchanges = subSet("all", [pair]);
+  Exchanges.open();
 
   // Code for parsing the maps in each exchange
   var mapParse = require("./../data-parsing/map-parsing.js");
 
-  // Condensing maps into the highest bid and lowest ask
-  var Krak_Info = mapParse(allExchangeData.kraken, "kraken");
-  var Polo_Info = mapParse(allExchangeData.poloniex, "poloniex");
-  var GDAX_Info = mapParse(allExchangeData.gdax, "gdax");
-  var Bitf_Info = mapParse(allExchangeData.bitfinex, "bitfinex");
-
-  // Creating a overall array to store highest bids and lowest asks
-  var overall = [Polo_Info, GDAX_Info, Bitf_Info, Krak_Info];
+  // Calling the actual arbitrage logic in the time interval specified
+  setInterval(function () {
+    for (var key in Exchanges) {
+      if (Exchanges.hasOwnProperty(key)) {
+        for (var key_2 in Exchanges[key]){
+          if (Exchanges[key].hasOwnProperty(key_2)) {
+            data[key] = mapParse(Exchanges[key][key_2], key);
+          }
+        }
```

Comment:
Also to clarify on point 1, this for loop specifically opens every form of data requested. My idea behind the sub-set class was to create a function call that would request only the data needed, and then using a double for loop, open all the specific order books for all the data requested.